pkg/gadgets/run/types/metadata.go

```diff
@@ -24,9 +24,6 @@ import (
 	"github.com/hashicorp/go-multierror"
 	log "github.com/sirupsen/logrus"
 
-	"github.com/inspektor-gadget/inspektor-gadget/pkg/btfhelpers"
```

Comment:
After these changes and a couple of changes in the metadata files, I think can revert https://github.com/inspektor-gadget/inspektor-gadget/commit/eb6de7b7a77c7353d9c24fca1cfc05d255deee23. Do you want to do it in another PR?

Reply:
I'd prefer to wait a bit until we understand how to implement it again.

Reply:
I saw you created the function to populate the datasource yaml leaf. I tested it with `--update-metadata` and it worked. I ensures now the alphabetical order so some yaml files need to be updated, but other than that, I don't see what is missing. What am I not considering?

Reply:
> I ensures now the alphabetical order so some yaml files need to be updated

This is what I was afraid of. Are you sure it's always the alphabetical order? If it's the case, we can re-enable that.

Reply:
I'm merging it right now to unblock other changes. I'll think about this later on.